app/api/courses.js

```diff
@@ -0,0 +1,119 @@
+    })
+
+})
+
+//User Creates a course
+//Authenticate token, create course & create enrollment for logged in user as teacher in the course
+router.post('/', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+
+    // create course & role as a teacher in enrollment
+    // create course 
+    if (req.body.name) { // name is required, other fields can be left unspecified
+        // create course with valid fields
+        const course = await db.Course.create(courseService.extractCourseFields(req.body))
+        
+        //create the enrollment
+        const enrollmentToInsert = {
+            courseId: course.id,
+            userId: user.id,
+            role: 'teacher'
+            // no section because they are a teacher
+        }
+        const missingFields = enrollmentService.validateEnrollmentCreationRequest(enrollmentToInsert)
+        if (missingFields.length == 0) {
+            const enrollment = await db.Enrollment.create(enrollmentToInsert)
+            res.status(201).send({
+                course: courseService.extractCourseFields(course),
+                enrollment: enrollmentService.extractEnrollmentFields(enrollment)
+            })
+        } else {
+            res.status(400).send({error: `Enrollment did not have all the required fields, it was missing ${missingFields}`})
+        }
+    } else {
+        res.status(400).send({error: "A course requires a name"})
+    }
+
+})
+
+//User joins with course code
+//Authenticate token & create enrollment for the user in the section that has a code 
+router.post('/join', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+
+    if (req.body.joinCode) {
+        const joinCode = req.body.joinCode
+
+        const sectionArray = await db.Section.findAll({
+            where: { joinCode: joinCode }
+        })
+
+        if (sectionArray && sectionArray.length > 0) {
+            const section = sectionArray[0]
+            // create the enrollment
+            const enrollmentToInsert = {
+                sectionId: section.id,
+                userId: user.id,
+                role: 'student'
+                // no course because they are a student
+            }
+            const missingFields = enrollmentService.validateEnrollmentCreationRequest(enrollmentToInsert)
+            if (missingFields.length == 0) {
+                const enrollment = await db.Enrollment.create(enrollmentToInsert)
+                res.status(201).send({
+                    section: sectionService.extractSectionFields(section),
+                    enrollment: enrollmentService.extractEnrollmentFields(enrollment)
+                })
+            } else {
+                res.status(400).send({error: `Enrollment did not have all the required fields, it was missing ${missingFields}`})
```

Comment:
It might make sense to use one of the string helpers here - I wrote one that serializes the array into csv